### PR TITLE
Added Gauss Seidel's algorithm

### PR DESCRIPTION
**Pull request by MUCCHU**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: linear_algebra/src/gauss_seidel.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# Gauss Seidel Method (https://en.wikipedia.org/wiki/Gauss%E2%80%93Seidel_method)
+# initially assumes the solution to be zero
+# Then in each iteration improves the solution by using the previous solution
+
+
+# Defining our function as gauss_seidel which takes 4 arguments
+# as A matrix(nxn), Solution and B matrix and max iterations
+
+
+def update_solution(a: list[list], x: list, b: list) -> list:
```
Comment (cclauss) [inline linear_algebra/src/gauss_seidel.py:12]:
```suggestion
def update_solution(matrix_a: list[list], solution: list, matrix_b: list) -> list:
```

File: linear_algebra/src/gauss_seidel.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    """
+    Function to update the solution
+    a: coefficient matrix
+    x: solution vector
+    b: constant vector
```
Comment (cclauss) [inline linear_algebra/src/gauss_seidel.py:17]:
```suggestion
```
Much better for the code to be readable than the comments.

**Comment by MUCCHU** [comment]
@cclauss I have made the changes. Kindly review and approve the Pull Request.

File: linear_algebra/src/gauss_seidel.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    [0.6, 2.3272727272727276, -0.9872727272727271, 0.8788636363636363]
+    """
+    n = len(coefficient_matrix)  # number of variables
+    for i in range(0, n):
+        # Update each variable
+        d = constant_vector[i]
+        for j in range(0, n):
+            if i != j:
+                d -= coefficient_matrix[i][j] * solution_vector[j]
+        solution_vector[i] = d / coefficient_matrix[i][i]
```
[the remark this answers is not in the document]
Comment (MUCCHU) [reply linear_algebra/src/gauss_seidel.py]: Done, Can you please review it.

**Pull request closed without merging**